ext/mbstring/mbstring.c

```diff
@@ -1158,6 +1158,7 @@ PHP_RSHUTDOWN_FUNCTION(mbstring)
 	MBSTRG(outconv_state) = 0;
 
 	if (MBSTRG(all_encodings_list)) {
+		GC_REMOVE_FROM_BUFFER(MBSTRG(all_encodings_list));
```

Comment:
We should actually use `zend_array_destroy` instead of `zend_hash_destroy`. It includes a call to `GC_REMOVE_FROM_BUFFER` and might be faster due to avoiding a dynamic call to `ht->pDestructor` and inlining `i_zval_ptr_dtor`. I'll adjust this.